### ISSUE TITLE
[plasma-desktop 5.18.5-2] Applet crashing rear opening Thermal Monitor Settings

### ISSUE THREAD

**Issue by hardwareadictos** [kotelnik/plasma-applet-thermal-monitor#67]
Rear updating my arch something broke the Widget. Done it on firstly on my laptop and did it two days later on my Desktop (same Arch installation on both) with the same results.

I'm going to upload a log and 

the packets that got updated (because i don't know which one caused the breakage).

[thermal_monitor_journalctl.txt](https://github.com/kotelnik/plasma-applet-thermal-monitor/files/4621702/thermal_monitor_journalctl.txt)

[package_update.txt](https://github.com/kotelnik/plasma-applet-thermal-monitor/files/4621717/package_update.txt)

I already tested other widgets and none of them are having issues.

Plasma version: plasma-desktop 5.18.5-2

Let me know if you need more info :)

**Comment by TimoWilken** [comment]
I installed `plasma5-applets-thermal-monitor` from the Arch Linux community repository (and nothing else; I had already installed its dependencies). Adding the applet to the plasma bar works, but trying to configure the applet crashes `plasmashell`, even after a reboot.

**Comment by hardwareadictos** [comment]
> I installed `plasma5-applets-thermal-monitor` from the Arch Linux community repository (and nothing else; I had already installed its dependencies). Adding the applet to the plasma bar works, but trying to configure the applet crashes `plasmashell`, even after a reboot.

That's my exact issue. I even tried to install it via KDE store and via KDE applet application.

**Comment by rohitghali** [comment]
Even I tried installing it from everywhere. Same issue. Best applet. Please fix it.

**Comment by imatimba** [comment]
Bug report on KDE's side: https://bugs.kde.org/show_bug.cgi?id=421392

**Comment by hardwareadictos** [comment]
> Bug report on KDE's side: https://bugs.kde.org/show_bug.cgi?id=421392

Nice! But I think that's not caused by Nvidia driver update. I suspect more about plasma lts update imo

**Comment by imatimba** [comment]
> > Bug report on KDE's side: https://bugs.kde.org/show_bug.cgi?id=421392
> 
> Nice! But I think that's not caused by Nvidia driver update. I suspect more about plasma lts update imo

Yeah, when I saw this bug report just now I did a new comment disregarding the info about the gpu/drivers

**Comment by Ashark** [comment]
There is also a [Fan Speed Monitor](https://github.com/dwardor/plasma-applet-fanspeed-monitor) applet, which says in Readme that it is based on this widget, and it also crashes when trying to configure widget.

**Comment by imatimba** [comment]
Well, the kde bug report was marked as duplicate of a generic 4 year old bug report that no one has even look at, it seems.
15 bug reports were marked as duplicates of this old bug report already. Not sure what to think about it. 
It looks like a WONTFIX to me.

**Comment by hardwareadictos** [comment]
> Well, the kde bug report was marked as duplicate of a generic 4 year old bug report that no one has even look at, it seems.
> 15 bug reports were marked as duplicates of this old bug report already. Not sure what to think about it.
> It looks like a WONTFIX to me.

Well, that's not a KDE issue, it's an applet issue. We won't get any support outside here...

**Comment by akien-mga** [comment]
> Well, that's not a KDE issue, it's an applet issue. We won't get any support outside here...

I beg to differ, if a plasmoid can crash plasmashell, that's a bug in plasmashell.

The issue can likely be worked around in the plasmoid to avoid making plasmashell crash, but the latter should not be crashable by user plugins in the first place.

**Comment by imatimba** [comment]
> > Well, the kde bug report was marked as duplicate of a generic 4 year old bug report that no one has even look at, it seems.
> > 15 bug reports were marked as duplicates of this old bug report already. Not sure what to think about it.
> > It looks like a WONTFIX to me.
> 
> Well, that's not a KDE issue, it's an applet issue. We won't get any support outside here...

Considering the applet was working fine until now it's possible that the bug is on KDE's side.  
Unless they've changed some internal behavior on purpose that makes code in this applet "the" bug.

edit: posted new comment by mistake.
I was thinking on doing a git bisect, but considering the complexity and size of KDE I wouldn't know where to start.
The bug could be on plasma-workspace, plasmashell, plasma-framework, etc.

**Comment by imatimba** [comment]
I was thinking on doing a git bisect, but considering the complexity and size of KDE I wouldn't know where to start.  
The bug could be on plasma-workspace, plasmashell, plasma-framework, etc.

**Comment by jnines** [comment]
For the time being I'm just using a command in the Command Output plasmoid to get my CPU/GPU temps.  I don't know if kotelnik is still active or not, or if one of the numerous forks has it fixed.

**Comment by acutbal** [comment]
Hi!!
I also have this issue on my computers, every time I try to config the thermal widget crashes plasma.
Regards.

**Comment by HayriGomer** [comment]
Same here , latest update broke it , so i think problem is plasma(desktop / framework / QT ... etc) not the applet itself  . Hopefully next update it will be fixed

**Comment by XinqiBao** [comment]
same issue here.
And I found another widget `kargos`, can run script and works really well for me.
Thanks @roachsinai shared this [solution](https://github.com/kotelnik/plasma-applet-thermal-monitor/issues/53#issuecomment-457652567), even it was for another issue.

**Comment by mariocarbone** [comment]
Is there any way to try to config it via a file?
I removed it and reinstalled apllied the fix and don't work anyway

This is my latest config
![image](https://user-images.githubusercontent.com/14996376/82928081-fe0bb880-9f81-11ea-9fa7-354db501aa34.png)

Now i have the shitty thermal icon and i can't open the config menu
 :(

**Comment by valdotdev** [comment]
Same problem on Manjaro 20.0. Crashes when trying to access the settings.

**Comment by mariocarbone** [comment]
> Same problem on Manjaro 20.0. Crashes when trying to access the settings.

Do you know how to configure it without menu?

**Comment by roachsinai** [comment]
> > Same problem on Manjaro 20.0. Crashes when trying to access the settings.
> 
> Do you know how to configure it without menu?

Should be in `~/.config/plasma-org.kde.plasma.desktop-appletsrc`.

**Comment by mruiz42** [comment]
> I beg to differ, if a plasmoid can crash plasmashell, that's a bug in plasmashell.
> 
> The issue can likely be worked around in the plasmoid to avoid making plasmashell crash, but the latter should not be crashable by user plugins in the first place.

Agreed. Here is the output for mine. It seems to crash right after detecting the disks for me. Not sure what the debug should look like when it opens the config window under normal conditions.
https://pastebin.com/D0AUaLPT

**Comment by hardwareadictos** [comment]
Hey! Today i updated my Fedora 32 Workstation and that bug appeared on KDE also.

**Comment by veganvelociraptor** [comment]
I have the same issue here using Arch and Manjaro. It can't be related to any nvidia driver issue, because i don't have any nvidia card nor drivers installed.

**Comment by BullShark** [comment]
> I installed `plasma5-applets-thermal-monitor` from the Arch Linux community repository (and nothing else; I had already installed its dependencies). Adding the applet to the plasma bar works, but trying to configure the applet crashes `plasmashell`, even after a reboot.

This issue also affects me in Manjaro. I tried compiling the source from git. I tried installing the AUR package. I had the same problems with both.

**Comment by bernharl** [comment]
Same here, on Manjaro. Installed the AUR package that was newly updated. It crashes my desktop when trying to convigure.

**Comment by jnines** [comment]
The problem is upstream [bug report](https://bugs.kde.org/show_bug.cgi?id=421392) and who knows if it'll be wontfix.  I've moved on to kargos, requires a little scripting but you do a lot more with it.

**Comment by moconnor1234** [comment]
@jnines would you mind sharing the script you used?

**Comment by ghost** [comment]
In the meantime, here's mine kargos one (I also use [icons-in-terminal](https://github.com/sebastiencs/icons-in-terminal) for the glyphs). You probably want to adapt yours, since I have my own named stuff in lm_sensors:

```
cputemp=$(sensors | grep 'CPU' | awk '{total += $3; count++} END {print total/count}')
gputemp=$(nvidia-smi --query-gpu=temperature.gpu --format=csv,noheader)
sdatemp=$(sudo /usr/bin/hddtemp /dev/sda | cut -c 34-35)
sdbtemp=$(sudo /usr/bin/hddtemp /dev/sdb | cut -c 31-32)

# Status
echo "<font size='2'>  ${cputemp}°&nbsp;  ${gputemp}°&nbsp;¹ ${sdatemp}°&nbsp;² ${sdbtemp}°| font= SF Pro Display:style=Medium size=14"

```

**Comment by jnines** [comment]
I use a few different scripts separated out for different things:
```
temp=$(sensors | grep -oP 'Tdie.*?\+\K[0-9.]+') 
echo "${temp%%.*}° | color=red bash='/usr/bin/ksysguard' onclick=bash"
```
for Intel you'd swap Tdie with Package.

```
gpu_temp=$(nvidia-smi --query-gpu=temperature.gpu --format=csv,noheader)
echo "${gpu_temp}° | color=green
```
for Nvidia.  I don't have an AMD GPU but I'm assuming it would show up under lm_sensors.

I also use a few others for things like mic, speakers, battery level for headset.

**Comment by tkalfaoglu** [comment]
For my AMD, I used kargos and this command:
sensors|grep edge|cut  -d'+' -f2|cut -d' ' -f1
